rqt_shell/src/rqt_shell/spyder_shell_widget.py

```diff
@@ -88,7 +90,12 @@ def create_process(self):
         self.process.finished.connect(self.finished)
         self.process.finished.connect(self.close_signal)
 
-        self.process.start('/bin/bash', ['-i'])
+        if scriptPath:
+            options = ["-c 'source " + os.path.abspath(scriptPath) + "; \
+                       /bin/bash -i'"])
```

Comment:
quoting the script path would get complicated really quick. Forward and back slashes, spaces, special characters, etc. all would need to be escaped properly.